rednose.py

```diff
@@ -366,7 +367,9 @@ def __init__(self, stream, excludes):
 		self.__stream = stream
 		self.__excludes = list(map(re.compile, excludes))
 
-	def __should_filter(self):
+	def __should_filter(self, a):
+		if a not in [('.',), ('X',), ('F',), ('S',)]:
+			return False
```

Comment:
so this means we _only ever_ filter output if it's a '.', 'X', 'F', or 'S'?

Wouldn't this change allow most from blacklisted sources? (e.g. long descriptions aren't going to be any of those 4 characters).

It sounds like you want a limited blacklist on just the nose timer source, without affecting the other sources. Would something like this work?:

```
filter_everything = lambda a: True
filter_test_status: lambda a: a in [('.',), ('X',), ('F',), ('S',)]
 BLACKLISTED_WRITERS = [
 	('nose[\\/]result\\.pyc?$', filter_everything),
 	('nose[\\/]result\\.pyc?$', filter_everything),
	('unittest[\\/]runner\\.pyc?$', filter_everything),
	('nosetimer[\\/]plugin\\.pyc?$', filter_test_status),
]
```

(with appropriate changes to the FilteringStream code to check `a` with the associated source-specific filter function)